lms/djangoapps/mobile_api/users/serializers.py

```diff
@@ -34,6 +35,10 @@ def to_representation(self, course_overview):
                 kwargs={'course_id': course_id},
                 request=request
             )
+            discussion_url = reverse(
+                'discussion_course',
+                kwargs={'course_id': course_id},
+                request=request) if course_overview.is_discussion_tab_enabled() else None
```

Comment:
ah! `UrlResetMixin` does the magic. Thank you @nasthagiri 
One last point; we only need to enable discussion api for test_discussion_url so I have only placed the patch on that test.